# Importing Libraries

In [1]:
import pandas as pd
import os # use this to access your environment variables
import requests # this will be used to call the APIs
import numpy as np

In [2]:
os.chdir('C:/Users/Laven/Documents/Data_Analysis/Capstone_Project')

## Joining Transit Stops

In [6]:
montreal_bike_stations = pd.read_csv("complete_citybikes_df.csv")

In [6]:
montreal_bike_stations

station_id  num_bikes_available  \
0             1                   12   
1             3                    5   
2            11                   11   
3            15                    8   
4            19                    2   
..          ...                  ...   
174         850                    2   
175         873                   16   
176         886                    8   
177         888                   13   
178         933                    0   

                                                name   Latitude  Longitude  \
0            Métro Champ-de-Mars (Viger / Sanguinet)  45.510253 -73.556777   
1                                    Clark / Ontario  45.510599 -73.566925   
2                   de la Porte de Québec / St-André  45.513294 -73.550578   
3                          St-Denis / de Maisonneuve  45.514339 -73.561685   
4               Métro Sherbrooke (de Rigaud / Berri)  45.518143 -73.568004   
..                                               ...        ...        ...   
174                            de Chambly / de Rouen  45.547103 -73.548924   
175                  Robert-Bourassa / René-Lévesque  45.502183 -73.568254   
176                         Molson / Wiliam-Tremblay  45.541603 -73.564961   
177  Complexe Desjardins (St-Urbain / René-Lévesque)  45.507885 -73.563151   
178                                             8178   0.000000   0.000000   

    Geohash  
0     f25dy  
1     f25dv  
2     f25dy  
3     f25dy  
4     f25dv  
..      ...  
174   f25en  
175   f25dv  
176   f25ej  
177   f25dy  
178   s0000  

[179 rows x 6 columns]

In [7]:
montreal_bike_stations = montreal_bike_stations.rename(columns={'name': 'Name'})

In [7]:
montreal_subway_stations = pd.read_csv("montreal_subway_stations3.csv")

In [9]:
montreal_subway_stations

Name                                            Address  \
0      Saint Laurent                   Montreal, Quebec H2X 1J5, Canada   
1               Peel                   Montreal, Quebec H3A 1T1, Canada   
2         Mont-Royal  470 Mont-Royal Ave E, Montreal, Quebec H2J 1W4...   
3     Place-des-Arts          Jeanne-Mance St, Montreal, Quebec, Canada   
4      Place-d'Armes                       Montreal, QC H2Z 1K4, Canada   
..               ...                                                ...   
63  Honoré-Beaugrand  Sherbrooke Street / rue Honoré-Beaugrand, Mont...   
64           Cartier  5 Boulevard Cartier Ouest, Laval, Quebec, H7N ...   
65    De La Concorde  1200 Boulevard de la Concorde Ouest, Laval, Qu...   
66       Montmorency  1800 Rue Jacques-Tétreault, Laval, Quebec, H7N...   
67            Acadie  6900, Boulevard de l'Acadie. Mont Royal, Montr...   

     Latitude  Longitude Geohash  
0   45.510852 -73.564836   f25dv  
1   45.500888 -73.574601   f25dv  
2   45.524534 -73.581672   f25dv  
3   45.507791 -73.568813   f25dv  
4   45.505763 -73.559876   f25dy  
..        ...        ...     ...  
63  45.593300 -73.524600   f25eq  
64  45.560100 -73.681700   f25e5  
65  45.560400 -73.710100   f25e4  
66  45.558400 -73.721500   f25e4  
67  45.523400 -73.623700   f25du  

[68 rows x 5 columns]

In [8]:
montreal_light_rail_stations = pd.read_csv("montreal_light_rail.csv")

In [11]:
montreal_light_rail_stations

Name  \
0                            Gare Centrale   
1                    Central Train Station   
2                           Place-des-Arts   
3                   Station Île-des-Soeurs   
4              Train Station Gare Centrale   
5                    Montréal West Station   
6               REM ÎLE-DES-SOEURS Station   
7                                   McGill   
8                             De Castelnau   
9                                    Sauvé   
10                                  Dorval   
11                                Ahuntsic   
12                               Angrignon   
13  VIA Rail - Montreal Maintenance Centre   
14         Central Station Complex-Homburg   
15                        VIA RAIL STATION   
16                             Bonaventure   
17                       Dorval (VIA Rail)   
18                   Terminus Centre-Ville   
19                             Montpellier   
20                         VIA Rail Canada   
21                              Bois-Franc   
22                     Station Du Quartier   
23                     Roxboro-Pierrefonds   
24                              Du Collège   

                                              Address   Latitude  Longitude  \
0   895 Rue De la Gauchetière O, Montréal, QC H3B ...  45.500112 -73.566522   
1   1001 Pl. Jean-Paul-Riopelle, Montréal, QC H2Z ...  45.499460 -73.564974   
2           Jeanne-Mance St, Montreal, Quebec, Canada  45.507791 -73.568813   
3                                Montreal, QC, Canada  45.470325 -73.538879   
4   935 Rue De la Gauchetière O, Montréal, QC H3B ...  45.499327 -73.566726   
5   7499 Ave Harley, Montreal, Quebec H4B 1L7, Canada  45.453446 -73.641674   
6    900 Bd René Lévesque, Verdun, QC H3E 1H5, Canada  45.470360 -73.538467   
7                        Montreal, QC H3A 2A4, Canada  45.504052 -73.571523   
8   1 Rue de Castelnau O, Montréal, QC H2R 2X9, Ca...  45.535437 -73.619873   
9                    Montreal, Quebec H3L 2B8, Canada  45.549552 -73.654346   
10                     Dorval, Quebec H9S 1A2, Canada  45.448751 -73.741314   
11  9615 Bd de l'Acadie, Montréal, QC H4N 1L8, Canada  45.535339 -73.662293   
12  3500 Trinitaires Blvd, Montreal, Quebec H4E 4J...  45.446323 -73.603773   
13          201 Av. Ash, Montréal, QC H3K 3K2, Canada  45.477232 -73.550840   
14  895 Rue De la Gauchetière O, Montréal, QC H3B ...  45.500113 -73.566523   
15  1110 Sherbrooke St W, Montreal, Quebec H3A 1G9...  45.500922 -73.568015   
16                       Montreal, QC H3B 4W5, Canada  45.498033 -73.566812   
17  755 Bd Montréal-Toronto, Dorval, QC H9S 1A1, C...  45.448697 -73.741511   
18                           Montreal, Quebec, Canada  45.498446 -73.566172   
19                       Montreal, QC H4N 1E8, Canada  45.524866 -73.672109   
20            895 Rue E, Montréal, QC H3B 2M4, Canada  45.500165 -73.565964   
21  5465 Henri Bourassa Blvd W, Saint-Laurent, Que...  45.522995 -73.709627   
22                               Brossard, QC, Canada  45.446857 -73.433517   
23        Dollard-Des Ormeaux, Quebec H8Y 2M2, Canada  45.509196 -73.809225   
24                       Montreal, QC H4L 2L7, Canada  45.509456 -73.674681   

   Geohash  
0    f25dv  
1    f25dv  
2    f25dv  
3    f25dw  
4    f25dv  
5    f25ds  
6    f25dw  
7    f25dv  
8    f25eh  
9    f25e5  
10   f25d9  
11   f25e5  
12   f25dt  
13   f25dw  
14   f25dv  
15   f25dv  
16   f25dv  
17   f25d9  
18   f25dv  
19   f25dg  
20   f25dv  
21   f25df  
22   f25f8  
23   f25db  
24   f25dg

In [9]:
#Preprare Null Columns For Concatenation

montreal_subway_stations['station_id'] = pd.NA
montreal_subway_stations['num_bikes_available'] = pd.NA

montreal_light_rail_stations['station_id'] = pd.NA
montreal_light_rail_stations['num_bikes_available'] = pd.NA


In [13]:
#Adding Types Before Concatenation
montreal_bike_stations['Type'] = 'Citybike'
montreal_subway_stations['Type'] = 'Subway'
montreal_light_rail_stations['Type'] = 'Light Rail'

In [14]:
montreal_transport_stations = pd.concat([
    montreal_bike_stations,
    montreal_subway_stations,
    montreal_light_rail_stations
])

In [15]:
montreal_transport_stations

station_id num_bikes_available                                     Name  \
0           1                  12  Métro Champ-de-Mars (Viger / Sanguinet)   
1           3                   5                          Clark / Ontario   
2          11                  11         de la Porte de Québec / St-André   
3          15                   8                St-Denis / de Maisonneuve   
4          19                   2     Métro Sherbrooke (de Rigaud / Berri)   
..        ...                 ...                                      ...   
20        NaN                 NaN                          VIA Rail Canada   
21        NaN                 NaN                               Bois-Franc   
22        NaN                 NaN                      Station Du Quartier   
23        NaN                 NaN                      Roxboro-Pierrefonds   
24        NaN                 NaN                               Du Collège   

     Latitude  Longitude Geohash        Type  \
0   45.510253 -73.556777   f25dy    Citybike   
1   45.510599 -73.566925   f25dv    Citybike   
2   45.513294 -73.550578   f25dy    Citybike   
3   45.514339 -73.561685   f25dy    Citybike   
4   45.518143 -73.568004   f25dv    Citybike   
..        ...        ...     ...         ...   
20  45.500165 -73.565964   f25dv  Light Rail   
21  45.522995 -73.709627   f25df  Light Rail   
22  45.446857 -73.433517   f25f8  Light Rail   
23  45.509196 -73.809225   f25db  Light Rail   
24  45.509456 -73.674681   f25dg  Light Rail   

                                              Address  
0                                                 NaN  
1                                                 NaN  
2                                                 NaN  
3                                                 NaN  
4                                                 NaN  
..                                                ...  
20            895 Rue E, Montréal, QC H3B 2M4, Canada  
21  5465 Henri Bourassa Blvd W, Saint-Laurent, Que...  
22                               Brossard, QC, Canada  
23        Dollard-Des Ormeaux, Quebec H8Y 2M2, Canada  
24                       Montreal, QC H4L 2L7, Canada  

[272 rows x 8 columns]

In [81]:
montreal_transport_stations.to_csv('montreal_transport_stations.csv', index=False)

NameError: name 'montreal_transport_stations' is not defined

In [10]:
montreal_transport_stations = pd.read_csv('montreal_transport_stations.csv')

In [32]:
montreal_all_rail_stations = pd.concat([
    montreal_subway_stations,
    montreal_light_rail_stations
])

In [33]:
montreal_all_rail_stations

Name                                            Address  \
0         Saint Laurent                   Montreal, Quebec H2X 1J5, Canada   
1                  Peel                   Montreal, Quebec H3A 1T1, Canada   
2            Mont-Royal  470 Mont-Royal Ave E, Montreal, Quebec H2J 1W4...   
3        Place-des-Arts          Jeanne-Mance St, Montreal, Quebec, Canada   
4         Place-d'Armes                       Montreal, QC H2Z 1K4, Canada   
..                  ...                                                ...   
20      VIA Rail Canada            895 Rue E, Montréal, QC H3B 2M4, Canada   
21           Bois-Franc  5465 Henri Bourassa Blvd W, Saint-Laurent, Que...   
22  Station Du Quartier                               Brossard, QC, Canada   
23  Roxboro-Pierrefonds        Dollard-Des Ormeaux, Quebec H8Y 2M2, Canada   
24           Du Collège                       Montreal, QC H4L 2L7, Canada   

     Latitude  Longitude Geohash station_id num_bikes_available        Type  
0   45.510852 -73.564836   f25dv        NaN                 NaN      Subway  
1   45.500888 -73.574601   f25dv        NaN                 NaN      Subway  
2   45.524534 -73.581672   f25dv        NaN                 NaN      Subway  
3   45.507791 -73.568813   f25dv        NaN                 NaN      Subway  
4   45.505763 -73.559876   f25dy        NaN                 NaN      Subway  
..        ...        ...     ...        ...                 ...         ...  
20  45.500165 -73.565964   f25dv        NaN                 NaN  Light Rail  
21  45.522995 -73.709627   f25df        NaN                 NaN  Light Rail  
22  45.446857 -73.433517   f25f8        NaN                 NaN  Light Rail  
23  45.509196 -73.809225   f25db        NaN                 NaN  Light Rail  
24  45.509456 -73.674681   f25dg        NaN                 NaN  Light Rail  

[93 rows x 8 columns]

In [34]:
montreal_all_rail_stations.to_csv('montreal_all_rail_stations.csv', index=False)

In [35]:
montreal_all_rail_stations

Name                                            Address  \
0         Saint Laurent                   Montreal, Quebec H2X 1J5, Canada   
1                  Peel                   Montreal, Quebec H3A 1T1, Canada   
2            Mont-Royal  470 Mont-Royal Ave E, Montreal, Quebec H2J 1W4...   
3        Place-des-Arts          Jeanne-Mance St, Montreal, Quebec, Canada   
4         Place-d'Armes                       Montreal, QC H2Z 1K4, Canada   
..                  ...                                                ...   
20      VIA Rail Canada            895 Rue E, Montréal, QC H3B 2M4, Canada   
21           Bois-Franc  5465 Henri Bourassa Blvd W, Saint-Laurent, Que...   
22  Station Du Quartier                               Brossard, QC, Canada   
23  Roxboro-Pierrefonds        Dollard-Des Ormeaux, Quebec H8Y 2M2, Canada   
24           Du Collège                       Montreal, QC H4L 2L7, Canada   

     Latitude  Longitude Geohash station_id num_bikes_available        Type  
0   45.510852 -73.564836   f25dv        NaN                 NaN      Subway  
1   45.500888 -73.574601   f25dv        NaN                 NaN      Subway  
2   45.524534 -73.581672   f25dv        NaN                 NaN      Subway  
3   45.507791 -73.568813   f25dv        NaN                 NaN      Subway  
4   45.505763 -73.559876   f25dy        NaN                 NaN      Subway  
..        ...        ...     ...        ...                 ...         ...  
20  45.500165 -73.565964   f25dv        NaN                 NaN  Light Rail  
21  45.522995 -73.709627   f25df        NaN                 NaN  Light Rail  
22  45.446857 -73.433517   f25f8        NaN                 NaN  Light Rail  
23  45.509196 -73.809225   f25db        NaN                 NaN  Light Rail  
24  45.509456 -73.674681   f25dg        NaN                 NaN  Light Rail  

[93 rows x 8 columns]

In [28]:
montreal_all_rail_stations = pd.read_csv('montreal_all_rail_stations.csv')

In [17]:
lat_lon_list_full_montreal = list(zip(montreal_transport_stations['Latitude'], montreal_transport_stations['Longitude']))

In [27]:
# Convert list to DataFrame
lat_lon_list_full_montreal = pd.DataFrame(lat_lon_list_full_montreal, columns=['Latitude', 'Longitude'])


In [31]:
# Save to CSV
lat_lon_list_full_montreal.to_csv('lat_lon_list_full_montreal', index=False)



## Joining Bars And Restaurants

In [3]:
bars_in_montreal_collected_unique = pd.read_csv('bars_in_montreal_collected_unique_utf8.csv')

In [46]:
bars_in_montreal_collected_unique

Name                                            Address  \
0              Pub St-Paul         124 Rue Saint-Paul E, Montréal, QC H2Y 1G6   
1              Piranha Bar  680 Saint-Catherine St W, Montreal, Quebec H3B...   
2            Patrick's Pub      1707 Saint Denis St, Montreal, Quebec H2X 3K4   
3    MVP Resto-Bar Sportif    200 St Catherine St E, Montreal, Quebec H2X 1L1   
4        Pub John Michaels      458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2   
..                     ...                                                ...   
518            M Resto-Bar       3625 Rue Talon, Sainte-Catherine, QC J5C 1T7   
519        BLVD Bar & Gril            100 Rue Kepler, Châteauguay, QC J6K 2X9   
520               Barbossa   3956 A St Laurent Blvd, Montreal, Quebec H2W 1Y3   
521   KARAOKE BAR 3 MINOTS     3812 St Laurent Blvd, Montreal, Quebec H2W 1X6   
522  Castello Resto Lounge             615 Bd Curé-Labelle, Laval, QC H7V 2T8   

      Latitude  Longitude  Rating  Price Level Geohash  
0    45.506944 -73.552500     3.9          2.0   f25dy  
1    45.502852 -73.569942     4.1          1.0   f25dv  
2    45.515570 -73.564079     4.2          2.0   f25dy  
3    45.511730 -73.561981     3.8          2.0   f25dy  
4    45.508011 -73.553953     4.0          2.0   f25dy  
..         ...        ...     ...          ...     ...  
518  45.387405 -73.553055     4.4          2.0   f25dn  
519  45.360540 -73.712815     4.5          2.0   f25d4  
520  45.516651 -73.578985     4.1          2.0   f25dv  
521  45.515533 -73.576472     3.4          1.0   f25dv  
522  45.537344 -73.742808     4.2          2.0   f25e1  

[523 rows x 7 columns]

In [47]:
restaurants_in_montreal_collected_unique = pd.read_csv('restaurants_in_montreal_collected_unique_utf8.csv')

In [48]:
restaurants_in_montreal_collected_unique

Name  \
0                                   Restaurant Branzino   
1             Bar et Restaurant La Catrina - Vieux-Port   
2                         COMMODORE restaurant Montréal   
3                                  Restaurant Keung Kee   
4                                    Beijing Restaurant   
...                                                 ...   
2767                  Les Enfants Terribles — Outremont   
2768                       Pho Bep Cuisine Vietnamienne   
2769                                       Yuukai sushi   
2770                           Pizzeria Magpie Mile End   
2771  Sushi Momo Le Comptoir - Takeout Counter Par C...   

                                                Address   Latitude  Longitude  \
0     50 René-Lévesque Blvd W, Montreal, Quebec H2Z 1A2  45.508147 -73.561650   
1      119 De la Commune St W, Montreal, Quebec H2Y 2C7  45.503820 -73.553983   
2     355 Blvd. De Maisonneuve Ouest, Montreal, Queb...  45.507151 -73.569114   
3      70 Rue De la Gauchetière O, Montréal, QC H2Z 1C1  45.507253 -73.560407   
4      92 Rue De la Gauchetière O, Montréal, QC H2Z 1C1  45.506930 -73.560560   
...                                                 ...        ...        ...   
2767        1257 Bernard Ave, Outremont, Quebec H2V 1V8  45.519924 -73.609135   
2768      102 Rue Saint-Viateur O, Montréal, QC H2T 2L1  45.524414 -73.600046   
2769            5407 Park Ave, Montreal, Quebec H2V 4G9  45.522069 -73.601220   
2770            16 Maguire St, Montreal, Quebec H2T 1B8  45.524703 -73.595522   
2771          16 Ave des Pins, Montreal, Quebec H2W 1N3  45.515266 -73.574704   

      Rating  Price Level Geohash  
0        3.9          NaN   f25dy  
1        4.1          2.0   f25dy  
2        3.9          NaN   f25dv  
3        3.9          2.0   f25dy  
4        4.1          2.0   f25dy  
...      ...          ...     ...  
2767     3.9          3.0   f25du  
2768     4.5          NaN   f25dv  
2769     4.5          2.0   f25dv  
2770     4.4          2.0   f25dv  
2771     4.4          NaN   f25dv  

[2772 rows x 7 columns]

In [49]:
# Add a new column 'Type' to each DataFrame
bars_in_montreal_collected_unique['Type'] = 'Bar'
restaurants_in_montreal_collected_unique['Type'] = 'Restaurant'

In [50]:
bars_in_montreal_collected_unique

Name                                            Address  \
0              Pub St-Paul         124 Rue Saint-Paul E, Montréal, QC H2Y 1G6   
1              Piranha Bar  680 Saint-Catherine St W, Montreal, Quebec H3B...   
2            Patrick's Pub      1707 Saint Denis St, Montreal, Quebec H2X 3K4   
3    MVP Resto-Bar Sportif    200 St Catherine St E, Montreal, Quebec H2X 1L1   
4        Pub John Michaels      458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2   
..                     ...                                                ...   
518            M Resto-Bar       3625 Rue Talon, Sainte-Catherine, QC J5C 1T7   
519        BLVD Bar & Gril            100 Rue Kepler, Châteauguay, QC J6K 2X9   
520               Barbossa   3956 A St Laurent Blvd, Montreal, Quebec H2W 1Y3   
521   KARAOKE BAR 3 MINOTS     3812 St Laurent Blvd, Montreal, Quebec H2W 1X6   
522  Castello Resto Lounge             615 Bd Curé-Labelle, Laval, QC H7V 2T8   

      Latitude  Longitude  Rating  Price Level Geohash Type  
0    45.506944 -73.552500     3.9          2.0   f25dy  Bar  
1    45.502852 -73.569942     4.1          1.0   f25dv  Bar  
2    45.515570 -73.564079     4.2          2.0   f25dy  Bar  
3    45.511730 -73.561981     3.8          2.0   f25dy  Bar  
4    45.508011 -73.553953     4.0          2.0   f25dy  Bar  
..         ...        ...     ...          ...     ...  ...  
518  45.387405 -73.553055     4.4          2.0   f25dn  Bar  
519  45.360540 -73.712815     4.5          2.0   f25d4  Bar  
520  45.516651 -73.578985     4.1          2.0   f25dv  Bar  
521  45.515533 -73.576472     3.4          1.0   f25dv  Bar  
522  45.537344 -73.742808     4.2          2.0   f25e1  Bar  

[523 rows x 8 columns]

In [51]:
restaurants_in_montreal_collected_unique

Name  \
0                                   Restaurant Branzino   
1             Bar et Restaurant La Catrina - Vieux-Port   
2                         COMMODORE restaurant Montréal   
3                                  Restaurant Keung Kee   
4                                    Beijing Restaurant   
...                                                 ...   
2767                  Les Enfants Terribles — Outremont   
2768                       Pho Bep Cuisine Vietnamienne   
2769                                       Yuukai sushi   
2770                           Pizzeria Magpie Mile End   
2771  Sushi Momo Le Comptoir - Takeout Counter Par C...   

                                                Address   Latitude  Longitude  \
0     50 René-Lévesque Blvd W, Montreal, Quebec H2Z 1A2  45.508147 -73.561650   
1      119 De la Commune St W, Montreal, Quebec H2Y 2C7  45.503820 -73.553983   
2     355 Blvd. De Maisonneuve Ouest, Montreal, Queb...  45.507151 -73.569114   
3      70 Rue De la Gauchetière O, Montréal, QC H2Z 1C1  45.507253 -73.560407   
4      92 Rue De la Gauchetière O, Montréal, QC H2Z 1C1  45.506930 -73.560560   
...                                                 ...        ...        ...   
2767        1257 Bernard Ave, Outremont, Quebec H2V 1V8  45.519924 -73.609135   
2768      102 Rue Saint-Viateur O, Montréal, QC H2T 2L1  45.524414 -73.600046   
2769            5407 Park Ave, Montreal, Quebec H2V 4G9  45.522069 -73.601220   
2770            16 Maguire St, Montreal, Quebec H2T 1B8  45.524703 -73.595522   
2771          16 Ave des Pins, Montreal, Quebec H2W 1N3  45.515266 -73.574704   

      Rating  Price Level Geohash        Type  
0        3.9          NaN   f25dy  Restaurant  
1        4.1          2.0   f25dy  Restaurant  
2        3.9          NaN   f25dv  Restaurant  
3        3.9          2.0   f25dy  Restaurant  
4        4.1          2.0   f25dy  Restaurant  
...      ...          ...     ...         ...  
2767     3.9          3.0   f25du  Restaurant  
2768     4.5          NaN   f25dv  Restaurant  
2769     4.5          2.0   f25dv  Restaurant  
2770     4.4          2.0   f25dv  Restaurant  
2771     4.4          NaN   f25dv  Restaurant  

[2772 rows x 8 columns]

In [59]:
# Concatenate the two dataframes
bars_and_restaurants_df = pd.concat([bars_in_montreal_collected_unique, restaurants_in_montreal_collected_unique], ignore_index=True)

In [60]:
bars_and_restaurants_df

Name  \
0                                           Pub St-Paul   
1                                           Piranha Bar   
2                                         Patrick's Pub   
3                                 MVP Resto-Bar Sportif   
4                                     Pub John Michaels   
...                                                 ...   
3290                  Les Enfants Terribles — Outremont   
3291                       Pho Bep Cuisine Vietnamienne   
3292                                       Yuukai sushi   
3293                           Pizzeria Magpie Mile End   
3294  Sushi Momo Le Comptoir - Takeout Counter Par C...   

                                                Address   Latitude  Longitude  \
0            124 Rue Saint-Paul E, Montréal, QC H2Y 1G6  45.506944 -73.552500   
1     680 Saint-Catherine St W, Montreal, Quebec H3B...  45.502852 -73.569942   
2         1707 Saint Denis St, Montreal, Quebec H2X 3K4  45.515570 -73.564079   
3       200 St Catherine St E, Montreal, Quebec H2X 1L1  45.511730 -73.561981   
4         458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2  45.508011 -73.553953   
...                                                 ...        ...        ...   
3290        1257 Bernard Ave, Outremont, Quebec H2V 1V8  45.519924 -73.609135   
3291      102 Rue Saint-Viateur O, Montréal, QC H2T 2L1  45.524414 -73.600046   
3292            5407 Park Ave, Montreal, Quebec H2V 4G9  45.522069 -73.601220   
3293            16 Maguire St, Montreal, Quebec H2T 1B8  45.524703 -73.595522   
3294          16 Ave des Pins, Montreal, Quebec H2W 1N3  45.515266 -73.574704   

      Rating  Price Level Geohash        Type  
0        3.9          2.0   f25dy         Bar  
1        4.1          1.0   f25dv         Bar  
2        4.2          2.0   f25dy         Bar  
3        3.8          2.0   f25dy         Bar  
4        4.0          2.0   f25dy         Bar  
...      ...          ...     ...         ...  
3290     3.9          3.0   f25du  Restaurant  
3291     4.5          NaN   f25dv  Restaurant  
3292     4.5          2.0   f25dv  Restaurant  
3293     4.4          2.0   f25dv  Restaurant  
3294     4.4          NaN   f25dv  Restaurant  

[3295 rows x 8 columns]

In [54]:
bars_and_restaurants_df = bars_and_restaurants_df.to_csv('bars_and_restaurants_df_utf8.csv', index=False, encoding='utf-8-sig')


In [4]:
bars_and_restaurants_df = pd.read_csv('bars_and_restaurants_df_utf8.csv')

In [5]:
bars_and_restaurants_df

Name  \
0                                           Pub St-Paul   
1                                           Piranha Bar   
2                                         Patrick's Pub   
3                                 MVP Resto-Bar Sportif   
4                                     Pub John Michaels   
...                                                 ...   
3290                  Les Enfants Terribles — Outremont   
3291                       Pho Bep Cuisine Vietnamienne   
3292                                       Yuukai sushi   
3293                           Pizzeria Magpie Mile End   
3294  Sushi Momo Le Comptoir - Takeout Counter Par C...   

                                                Address   Latitude  Longitude  \
0            124 Rue Saint-Paul E, Montréal, QC H2Y 1G6  45.506944 -73.552500   
1     680 Saint-Catherine St W, Montreal, Quebec H3B...  45.502852 -73.569942   
2         1707 Saint Denis St, Montreal, Quebec H2X 3K4  45.515570 -73.564079   
3       200 St Catherine St E, Montreal, Quebec H2X 1L1  45.511730 -73.561981   
4         458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2  45.508011 -73.553953   
...                                                 ...        ...        ...   
3290        1257 Bernard Ave, Outremont, Quebec H2V 1V8  45.519924 -73.609135   
3291      102 Rue Saint-Viateur O, Montréal, QC H2T 2L1  45.524414 -73.600046   
3292            5407 Park Ave, Montreal, Quebec H2V 4G9  45.522069 -73.601220   
3293            16 Maguire St, Montreal, Quebec H2T 1B8  45.524703 -73.595522   
3294          16 Ave des Pins, Montreal, Quebec H2W 1N3  45.515266 -73.574704   

      Rating  Price Level Geohash        Type  
0        3.9          2.0   f25dy         Bar  
1        4.1          1.0   f25dv         Bar  
2        4.2          2.0   f25dy         Bar  
3        3.8          2.0   f25dy         Bar  
4        4.0          2.0   f25dy         Bar  
...      ...          ...     ...         ...  
3290     3.9          3.0   f25du  Restaurant  
3291     4.5          NaN   f25dv  Restaurant  
3292     4.5          2.0   f25dv  Restaurant  
3293     4.4          2.0   f25dv  Restaurant  
3294     4.4          NaN   f25dv  Restaurant  

[3295 rows x 8 columns]

In [61]:
#Check for places listed as both
columns_to_check = [col for col in bars_and_restaurants_df.columns if col != 'Type']



In [62]:
duplicate_rows = bars_and_restaurants_df.duplicated(subset=columns_to_check, keep=False)

duplicates_df = bars_and_restaurants_df[duplicate_rows]

In [63]:
duplicates_df

Name  \
19                      Winnie's Bar Restaurant   
25                          Bevo Bar + Pizzeria   
40    Bar et Restaurant La Catrina - Vieux-Port   
107                    Restaurant & Bar Le Kube   
137                        Réservoir - Brasseur   
...                                         ...   
2758                 Firegrill Ville St-Laurent   
2977                           Lord William Pub   
3007                         Station des Sports   
3008                            Le White Rabbit   
3105                                 Entre-Deux   

                                                Address   Latitude  Longitude  \
19           1455 Crescent St, Montreal, Quebec H3G 2B2  45.497962 -73.576909   
25           410 R. Saint-Vincent, Montréal, QC H2Y 3A5  45.507020 -73.553286   
40     119 De la Commune St W, Montreal, Quebec H2Y 2C7  45.503820 -73.553983   
107   Restaurant Hôtel, 900 Rue De la Gauchetière O,...  45.499151 -73.565686   
137                9 Av. Duluth E, Montréal, QC H2W 1G7  45.517265 -73.579263   
...                                                 ...        ...        ...   
2758  3500 Boulevard Cote Vertu Ouest, Saint-Laurent...  45.493860 -73.704339   
2977        265 Rue des Seigneurs, Montréal, QC H3J 0B3  45.486899 -73.567036   
3007     5773 Taschereau Blvd, Brossard, Quebec J4Z 1A5  45.478816 -73.464314   
3008         2151 Boul. Lapinière, Brossard, QC J4W 2T5  45.471927 -73.470309   
3105  5930 Sherbrooke St W Unite 102, Montreal, Queb...  45.468640 -73.619166   

      Rating  Price Level Geohash        Type  
19       3.9          2.0   f25dv         Bar  
25       4.3          2.0   f25dy         Bar  
40       4.1          2.0   f25dy         Bar  
107      3.4          NaN   f25dv         Bar  
137      4.5          2.0   f25dv         Bar  
...      ...          ...     ...         ...  
2758     4.0          2.0   f25df  Restaurant  
2977     4.3          2.0   f25dv  Restaurant  
3007     3.7          2.0   f25f8  Restaurant  
3008     4.0          2.0   f25f8  Restaurant  
3105     4.7          NaN   f25ds  Restaurant  

[104 rows x 8 columns]

In [64]:
duplicates_df.to_csv('duplicates_df_utf8.csv', index=False, encoding='utf-8-sig')

In [65]:
bars_and_restaurants_df_no_duplicates = bars_and_restaurants_df.drop_duplicates(subset=columns_to_check, keep=False)


In [66]:
bars_and_restaurants_df_no_duplicates

Name  \
0                                           Pub St-Paul   
1                                           Piranha Bar   
2                                         Patrick's Pub   
3                                 MVP Resto-Bar Sportif   
4                                     Pub John Michaels   
...                                                 ...   
3290                  Les Enfants Terribles — Outremont   
3291                       Pho Bep Cuisine Vietnamienne   
3292                                       Yuukai sushi   
3293                           Pizzeria Magpie Mile End   
3294  Sushi Momo Le Comptoir - Takeout Counter Par C...   

                                                Address   Latitude  Longitude  \
0            124 Rue Saint-Paul E, Montréal, QC H2Y 1G6  45.506944 -73.552500   
1     680 Saint-Catherine St W, Montreal, Quebec H3B...  45.502852 -73.569942   
2         1707 Saint Denis St, Montreal, Quebec H2X 3K4  45.515570 -73.564079   
3       200 St Catherine St E, Montreal, Quebec H2X 1L1  45.511730 -73.561981   
4         458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2  45.508011 -73.553953   
...                                                 ...        ...        ...   
3290        1257 Bernard Ave, Outremont, Quebec H2V 1V8  45.519924 -73.609135   
3291      102 Rue Saint-Viateur O, Montréal, QC H2T 2L1  45.524414 -73.600046   
3292            5407 Park Ave, Montreal, Quebec H2V 4G9  45.522069 -73.601220   
3293            16 Maguire St, Montreal, Quebec H2T 1B8  45.524703 -73.595522   
3294          16 Ave des Pins, Montreal, Quebec H2W 1N3  45.515266 -73.574704   

      Rating  Price Level Geohash        Type  
0        3.9          2.0   f25dy         Bar  
1        4.1          1.0   f25dv         Bar  
2        4.2          2.0   f25dy         Bar  
3        3.8          2.0   f25dy         Bar  
4        4.0          2.0   f25dy         Bar  
...      ...          ...     ...         ...  
3290     3.9          3.0   f25du  Restaurant  
3291     4.5          NaN   f25dv  Restaurant  
3292     4.5          2.0   f25dv  Restaurant  
3293     4.4          2.0   f25dv  Restaurant  
3294     4.4          NaN   f25dv  Restaurant  

[3191 rows x 8 columns]

In [67]:
bars_and_restaurants_df_no_duplicates.to_csv('bars_and_restaurants_df_no_duplicates_utf8.csv', index=False, encoding='utf-8-sig')



Copy pasting what I wrote in Microsoft word here as a quality control check when cleaning my complete data frame:

Number of entries in each table regarding bars and restaurants:

bars_and_restaurants_df_utf8 <-- 3295 rows not including header

bars_and_restaurants_df_no_duplicates_utf8 <-- 3192 rows not including header

duplicates_df_utf8 <-- 52 rows not including header after they have been boiled down to one row per dupilicate.

3192 + 2*(52) = 3295 so I have the right number of rows.


## Loading Complete DataFrame





In [13]:
bars_and_restaurants_complete = pd.read_csv('bars_and_restaurants_complete_utf8.csv')



In [14]:
bars_and_restaurants_complete



Name  \
0                                      Pub St-Paul   
1                                      Piranha Bar   
2                                    Patrick's Pub   
3                            MVP Resto-Bar Sportif   
4                                Pub John Michaels   
...                                            ...   
3238                        Le Barrage - Brasseurs   
3239                    Firegrill Ville St-Laurent   
3240  Wild Wing Brasserie Sportive Greenfield Park   
3241                                    Entre-Deux   
3242                               Le White Rabbit   

                                                Address   Latitude  Longitude  \
0            124 Rue Saint-Paul E, Montréal, QC H2Y 1G6  45.506944 -73.552500   
1     680 Saint-Catherine St W,  Montréal, Quebec H3...  45.502852 -73.569942   
2        1707 Saint Denis St,  Montréal, Quebec H2X 3K4  45.515570 -73.564079   
3      200 St Catherine St E,  Montréal, Quebec H2X 1L1  45.511730 -73.561981   
4         458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2  45.508011 -73.553953   
...                                                 ...        ...        ...   
3238    4200 Ch. de la Savane, Saint-Hubert, QC J3Y 9G5  45.512768 -73.432220   
3239  3500 Boulevard Cote Vertu Ouest, Saint-Laurent...  45.493860 -73.704339   
3240  3320 Taschereau Blvd, Greenfield Park, Quebec ...  45.493906 -73.481279   
3241  5930 Sherbrooke St W Unite 102,  Montréal, Que...  45.468640 -73.619166   
3242         2151 Boul. Lapinière, Brossard, QC J4W 2T5  45.471927 -73.470309   

      Rating  Price Level Geohash            Type  
0        3.9          2.0   f25dy             Bar  
1        4.1          1.0   f25dv             Bar  
2        4.2          2.0   f25dy             Bar  
3        3.8          2.0   f25dy             Bar  
4        4.0          2.0   f25dy             Bar  
...      ...          ...     ...             ...  
3238     4.7          2.0   f25fc  Bar/Restaurant  
3239     4.0          2.0   f25df  Bar/Restaurant  
3240     3.9          2.0   f25dz  Bar/Restaurant  
3241     4.7          NaN   f25ds  Bar/Restaurant  
3242     4.0          2.0   f25f8  Bar/Restaurant  

[3243 rows x 8 columns]

## Making GeoHash data frame


In [82]:
# Load the dataframe from the CSV file
montreal_transport_stations = pd.read_csv('montreal_transport_stations.csv')


In [86]:
# Load the dataframe from the CSV file
montreal_all_rail_stations = pd.read_csv('montreal_all_rail_stations.csv')

In [91]:
# Select only the 'Geohash' and 'Type' columns from each dataframe
stations_subset = montreal_transport_stations[['Geohash', 'Type', 'Address', 'Name']]
bars_restaurants_subset = bars_and_restaurants_complete[['Geohash', 'Type', 'Address', 'Name']]


In [92]:
rail_subset = montreal_all_rail_stations[['Geohash', 'Type', 'Address', 'Name']]

In [93]:
geohash_df = pd.concat([stations_subset, bars_restaurants_subset], ignore_index=True)


In [100]:
geohash_df_no_bikes = pd.concat([rail_subset, bars_restaurants_subset], ignore_index=True)




In [96]:
geohash_df.to_csv('geohash_df_utf8.csv', index=False, encoding='utf-8-sig')


In [102]:
geohash_df_no_bikes.to_csv('geohash_df_no_bikes_utf8.csv', index=False, encoding='utf-8-sig')




Quality control check after making GeoHash data frames:

bars_restaurants_complete has 3243 rows not including the header

montreal transport stations has 272 rows not including the header

geohash_df has 3515 rows not including the header

3243 + 272  = 3515 rows so this makes sense.

montreal_all_rail_stations has 93 rows not including the header

geohash_df_no_bikes has 3336 rows not including the header

3243 + 93 = 3336 rows so this makes sense.

Making More detailed GeoHashes

In [11]:
montreal_transport_stations

station_id  num_bikes_available                                     Name  \
0           1.0                 12.0  Métro Champ-de-Mars (Viger / Sanguinet)   
1           3.0                  5.0                          Clark / Ontario   
2          11.0                 11.0         de la Porte de Québec / St-André   
3          15.0                  8.0                St-Denis / de Maisonneuve   
4          19.0                  2.0     Métro Sherbrooke (de Rigaud / Berri)   
..          ...                  ...                                      ...   
267         NaN                  NaN                          VIA Rail Canada   
268         NaN                  NaN                               Bois-Franc   
269         NaN                  NaN                      Station Du Quartier   
270         NaN                  NaN                      Roxboro-Pierrefonds   
271         NaN                  NaN                               Du Collège   

      Latitude  Longitude Geohash        Type  \
0    45.510253 -73.556777   f25dy    Citybike   
1    45.510599 -73.566925   f25dv    Citybike   
2    45.513294 -73.550578   f25dy    Citybike   
3    45.514339 -73.561685   f25dy    Citybike   
4    45.518143 -73.568004   f25dv    Citybike   
..         ...        ...     ...         ...   
267  45.500165 -73.565964   f25dv  Light Rail   
268  45.522995 -73.709627   f25df  Light Rail   
269  45.446857 -73.433517   f25f8  Light Rail   
270  45.509196 -73.809225   f25db  Light Rail   
271  45.509456 -73.674681   f25dg  Light Rail   

                                               Address  
0                                                  NaN  
1                                                  NaN  
2                                                  NaN  
3                                                  NaN  
4                                                  NaN  
..                                                 ...  
267            895 Rue E, Montréal, QC H3B 2M4, Canada  
268  5465 Henri Bourassa Blvd W, Saint-Laurent, Que...  
269                               Brossard, QC, Canada  
270        Dollard-Des Ormeaux, Quebec H8Y 2M2, Canada  
271                       Montreal, QC H4L 2L7, Canada  

[272 rows x 8 columns]

In [15]:
bars_and_restaurants_complete



Name  \
0                                      Pub St-Paul   
1                                      Piranha Bar   
2                                    Patrick's Pub   
3                            MVP Resto-Bar Sportif   
4                                Pub John Michaels   
...                                            ...   
3238                        Le Barrage - Brasseurs   
3239                    Firegrill Ville St-Laurent   
3240  Wild Wing Brasserie Sportive Greenfield Park   
3241                                    Entre-Deux   
3242                               Le White Rabbit   

                                                Address   Latitude  Longitude  \
0            124 Rue Saint-Paul E, Montréal, QC H2Y 1G6  45.506944 -73.552500   
1     680 Saint-Catherine St W,  Montréal, Quebec H3...  45.502852 -73.569942   
2        1707 Saint Denis St,  Montréal, Quebec H2X 3K4  45.515570 -73.564079   
3      200 St Catherine St E,  Montréal, Quebec H2X 1L1  45.511730 -73.561981   
4         458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2  45.508011 -73.553953   
...                                                 ...        ...        ...   
3238    4200 Ch. de la Savane, Saint-Hubert, QC J3Y 9G5  45.512768 -73.432220   
3239  3500 Boulevard Cote Vertu Ouest, Saint-Laurent...  45.493860 -73.704339   
3240  3320 Taschereau Blvd, Greenfield Park, Quebec ...  45.493906 -73.481279   
3241  5930 Sherbrooke St W Unite 102,  Montréal, Que...  45.468640 -73.619166   
3242         2151 Boul. Lapinière, Brossard, QC J4W 2T5  45.471927 -73.470309   

      Rating  Price Level Geohash            Type  
0        3.9          2.0   f25dy             Bar  
1        4.1          1.0   f25dv             Bar  
2        4.2          2.0   f25dy             Bar  
3        3.8          2.0   f25dy             Bar  
4        4.0          2.0   f25dy             Bar  
...      ...          ...     ...             ...  
3238     4.7          2.0   f25fc  Bar/Restaurant  
3239     4.0          2.0   f25df  Bar/Restaurant  
3240     3.9          2.0   f25dz  Bar/Restaurant  
3241     4.7          NaN   f25ds  Bar/Restaurant  
3242     4.0          2.0   f25f8  Bar/Restaurant  

[3243 rows x 8 columns]

In [21]:
import geohash


In [22]:
# Define a function that takes latitude and longitude and returns a six-character geohash
def generate_geohash6(lat, lon):
    return geohash.encode(lat, lon, precision=6)

In [24]:
# Add precision 6 geohashes
montreal_transport_stations['Geohash_prec6'] = montreal_transport_stations.apply(lambda row: generate_geohash6(row['Latitude'], row['Longitude']), axis=1)

In [25]:
montreal_transport_stations

station_id  num_bikes_available                                     Name  \
0           1.0                 12.0  Métro Champ-de-Mars (Viger / Sanguinet)   
1           3.0                  5.0                          Clark / Ontario   
2          11.0                 11.0         de la Porte de Québec / St-André   
3          15.0                  8.0                St-Denis / de Maisonneuve   
4          19.0                  2.0     Métro Sherbrooke (de Rigaud / Berri)   
..          ...                  ...                                      ...   
267         NaN                  NaN                          VIA Rail Canada   
268         NaN                  NaN                               Bois-Franc   
269         NaN                  NaN                      Station Du Quartier   
270         NaN                  NaN                      Roxboro-Pierrefonds   
271         NaN                  NaN                               Du Collège   

      Latitude  Longitude Geohash        Type  \
0    45.510253 -73.556777   f25dy    Citybike   
1    45.510599 -73.566925   f25dv    Citybike   
2    45.513294 -73.550578   f25dy    Citybike   
3    45.514339 -73.561685   f25dy    Citybike   
4    45.518143 -73.568004   f25dv    Citybike   
..         ...        ...     ...         ...   
267  45.500165 -73.565964   f25dv  Light Rail   
268  45.522995 -73.709627   f25df  Light Rail   
269  45.446857 -73.433517   f25f8  Light Rail   
270  45.509196 -73.809225   f25db  Light Rail   
271  45.509456 -73.674681   f25dg  Light Rail   

                                               Address Geohash_prec6  
0                                                  NaN        f25dyh  
1                                                  NaN        f25dvu  
2                                                  NaN        f25dym  
3                                                  NaN        f25dyj  
4                                                  NaN        f25dvy  
..                                                 ...           ...  
267            895 Rue E, Montréal, QC H3B 2M4, Canada        f25dvg  
268  5465 Henri Bourassa Blvd W, Saint-Laurent, Que...        f25dfx  
269                               Brossard, QC, Canada        f25f8c  
270        Dollard-Des Ormeaux, Quebec H8Y 2M2, Canada        f25dbk  
271                       Montreal, QC H4L 2L7, Canada        f25dgk  

[272 rows x 9 columns]

In [29]:
montreal_all_rail_stations['Geohash_prec6'] = montreal_all_rail_stations.apply(lambda row: generate_geohash6(row['Latitude'], row['Longitude']), axis=1)

In [30]:
montreal_all_rail_stations

Name                                            Address  \
0         Saint Laurent                   Montreal, Quebec H2X 1J5, Canada   
1                  Peel                   Montreal, Quebec H3A 1T1, Canada   
2            Mont-Royal  470 Mont-Royal Ave E, Montreal, Quebec H2J 1W4...   
3        Place-des-Arts          Jeanne-Mance St, Montreal, Quebec, Canada   
4         Place-d'Armes                       Montreal, QC H2Z 1K4, Canada   
..                  ...                                                ...   
88      VIA Rail Canada            895 Rue E, Montréal, QC H3B 2M4, Canada   
89           Bois-Franc  5465 Henri Bourassa Blvd W, Saint-Laurent, Que...   
90  Station Du Quartier                               Brossard, QC, Canada   
91  Roxboro-Pierrefonds        Dollard-Des Ormeaux, Quebec H8Y 2M2, Canada   
92           Du Collège                       Montreal, QC H4L 2L7, Canada   

     Latitude  Longitude Geohash  station_id  num_bikes_available        Type  \
0   45.510852 -73.564836   f25dv         NaN                  NaN      Subway   
1   45.500888 -73.574601   f25dv         NaN                  NaN      Subway   
2   45.524534 -73.581672   f25dv         NaN                  NaN      Subway   
3   45.507791 -73.568813   f25dv         NaN                  NaN      Subway   
4   45.505763 -73.559876   f25dy         NaN                  NaN      Subway   
..        ...        ...     ...         ...                  ...         ...   
88  45.500165 -73.565964   f25dv         NaN                  NaN  Light Rail   
89  45.522995 -73.709627   f25df         NaN                  NaN  Light Rail   
90  45.446857 -73.433517   f25f8         NaN                  NaN  Light Rail   
91  45.509196 -73.809225   f25db         NaN                  NaN  Light Rail   
92  45.509456 -73.674681   f25dg         NaN                  NaN  Light Rail   

   Geohash_prec6  
0         f25dvu  
1         f25dvg  
2         f25dvx  
3         f25dvu  
4         f25dyh  
..           ...  
88        f25dvg  
89        f25dfx  
90        f25f8c  
91        f25dbk  
92        f25dgk  

[93 rows x 9 columns]

In [31]:
bars_and_restaurants_complete['Geohash_prec6'] = bars_and_restaurants_complete.apply(lambda row: generate_geohash6(row['Latitude'], row['Longitude']), axis=1)

In [32]:
bars_and_restaurants_complete

Name  \
0                                      Pub St-Paul   
1                                      Piranha Bar   
2                                    Patrick's Pub   
3                            MVP Resto-Bar Sportif   
4                                Pub John Michaels   
...                                            ...   
3238                        Le Barrage - Brasseurs   
3239                    Firegrill Ville St-Laurent   
3240  Wild Wing Brasserie Sportive Greenfield Park   
3241                                    Entre-Deux   
3242                               Le White Rabbit   

                                                Address   Latitude  Longitude  \
0            124 Rue Saint-Paul E, Montréal, QC H2Y 1G6  45.506944 -73.552500   
1     680 Saint-Catherine St W,  Montréal, Quebec H3...  45.502852 -73.569942   
2        1707 Saint Denis St,  Montréal, Quebec H2X 3K4  45.515570 -73.564079   
3      200 St Catherine St E,  Montréal, Quebec H2X 1L1  45.511730 -73.561981   
4         458 Pl. Jacques-Cartier, Montréal, QC H2Y 3Z2  45.508011 -73.553953   
...                                                 ...        ...        ...   
3238    4200 Ch. de la Savane, Saint-Hubert, QC J3Y 9G5  45.512768 -73.432220   
3239  3500 Boulevard Cote Vertu Ouest, Saint-Laurent...  45.493860 -73.704339   
3240  3320 Taschereau Blvd, Greenfield Park, Quebec ...  45.493906 -73.481279   
3241  5930 Sherbrooke St W Unite 102,  Montréal, Que...  45.468640 -73.619166   
3242         2151 Boul. Lapinière, Brossard, QC J4W 2T5  45.471927 -73.470309   

      Rating  Price Level Geohash            Type Geohash_prec6  
0        3.9          2.0   f25dy             Bar        f25dyk  
1        4.1          1.0   f25dv             Bar        f25dvg  
2        4.2          2.0   f25dy             Bar        f25dyj  
3        3.8          2.0   f25dy             Bar        f25dyj  
4        4.0          2.0   f25dy             Bar        f25dyh  
...      ...          ...     ...             ...           ...  
3238     4.7          2.0   f25fc  Bar/Restaurant        f25fcj  
3239     4.0          2.0   f25df  Bar/Restaurant        f25dfc  
3240     3.9          2.0   f25dz  Bar/Restaurant        f25dzc  
3241     4.7          NaN   f25ds  Bar/Restaurant        f25dsv  
3242     4.0          2.0   f25f8  Bar/Restaurant        f25f8j  

[3243 rows x 9 columns]

In [33]:
# Select only the 'Geohash' and 'Type' columns from each dataframe
stations_subset = montreal_transport_stations[['Geohash', 'Geohash_prec6', 'Type', 'Address', 'Name']]
bars_restaurants_subset = bars_and_restaurants_complete[['Geohash', 'Geohash_prec6', 'Type', 'Address', 'Name']]
rail_subset = montreal_all_rail_stations[['Geohash', 'Geohash_prec6', 'Type', 'Address', 'Name']]

In [34]:
geohash_6_df = pd.concat([stations_subset, bars_restaurants_subset], ignore_index=True)

In [35]:
geohash_6_no_bikes_df = pd.concat([stations_subset, bars_restaurants_subset], ignore_index=True)

In [36]:
geohash_6_df


Geohash Geohash_prec6            Type  \
0      f25dy        f25dyh        Citybike   
1      f25dv        f25dvu        Citybike   
2      f25dy        f25dym        Citybike   
3      f25dy        f25dyj        Citybike   
4      f25dv        f25dvy        Citybike   
...      ...           ...             ...   
3510   f25fc        f25fcj  Bar/Restaurant   
3511   f25df        f25dfc  Bar/Restaurant   
3512   f25dz        f25dzc  Bar/Restaurant   
3513   f25ds        f25dsv  Bar/Restaurant   
3514   f25f8        f25f8j  Bar/Restaurant   

                                                Address  \
0                                                   NaN   
1                                                   NaN   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
3510    4200 Ch. de la Savane, Saint-Hubert, QC J3Y 9G5   
3511  3500 Boulevard Cote Vertu Ouest, Saint-Laurent...   
3512  3320 Taschereau Blvd, Greenfield Park, Quebec ...   
3513  5930 Sherbrooke St W Unite 102,  Montréal, Que...   
3514         2151 Boul. Lapinière, Brossard, QC J4W 2T5   

                                              Name  
0          Métro Champ-de-Mars (Viger / Sanguinet)  
1                                  Clark / Ontario  
2                 de la Porte de Québec / St-André  
3                        St-Denis / de Maisonneuve  
4             Métro Sherbrooke (de Rigaud / Berri)  
...                                            ...  
3510                        Le Barrage - Brasseurs  
3511                    Firegrill Ville St-Laurent  
3512  Wild Wing Brasserie Sportive Greenfield Park  
3513                                    Entre-Deux  
3514                               Le White Rabbit  

[3515 rows x 5 columns]

In [38]:
geohash_list = ['f25dv', 'f25dy', 'f25ej']


In [39]:
geohash_6_df = geohash_6_df[geohash_6_df['Geohash'].isin(geohash_list)]

In [47]:
geohash_6_df

Geohash Geohash_prec6            Type  \
0      f25dy        f25dyh        Citybike   
1      f25dv        f25dvu        Citybike   
2      f25dy        f25dym        Citybike   
3      f25dy        f25dyj        Citybike   
4      f25dv        f25dvy        Citybike   
...      ...           ...             ...   
3485   f25dv        f25dvp  Bar/Restaurant   
3486   f25ej        f25eju  Bar/Restaurant   
3487   f25ej        f25ejg  Bar/Restaurant   
3490   f25dv        f25dvx  Bar/Restaurant   
3491   f25dy        f25dy5  Bar/Restaurant   

                                          Address  \
0                                             NaN   
1                                             NaN   
2                                             NaN   
3                                             NaN   
4                                             NaN   
...                                           ...   
3485     5834 Park Ave,  Montréal, Quebec H2V 4H3   
3486        3560 Rue Masson, Montréal, QC H1X 1R9   
3487        2800 Rue Masson, Montréal, QC H1Y 1W7   
3490       500 Rue Rachel E, Montréal, QC H2J 2H2   
3491  390 Rue Saint-Jacques, Montréal, QC H2Y 1S1   

                                         Name  
0     Métro Champ-de-Mars (Viger / Sanguinet)  
1                             Clark / Ontario  
2            de la Porte de Québec / St-André  
3                   St-Denis / de Maisonneuve  
4        Métro Sherbrooke (de Rigaud / Berri)  
...                                       ...  
3485             Mezcaleros Tapas & Cocktails  
3486                          Café Bar U-Turn  
3487                                Quai No.4  
3490                        Name's on the way  
3491          Bord'Elle Boutique Bar & Eatery  

[1204 rows x 5 columns]

In [37]:
geohash_6_no_bikes_df

Geohash Geohash_prec6            Type  \
0      f25dy        f25dyh        Citybike   
1      f25dv        f25dvu        Citybike   
2      f25dy        f25dym        Citybike   
3      f25dy        f25dyj        Citybike   
4      f25dv        f25dvy        Citybike   
...      ...           ...             ...   
3510   f25fc        f25fcj  Bar/Restaurant   
3511   f25df        f25dfc  Bar/Restaurant   
3512   f25dz        f25dzc  Bar/Restaurant   
3513   f25ds        f25dsv  Bar/Restaurant   
3514   f25f8        f25f8j  Bar/Restaurant   

                                                Address  \
0                                                   NaN   
1                                                   NaN   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
3510    4200 Ch. de la Savane, Saint-Hubert, QC J3Y 9G5   
3511  3500 Boulevard Cote Vertu Ouest, Saint-Laurent...   
3512  3320 Taschereau Blvd, Greenfield Park, Quebec ...   
3513  5930 Sherbrooke St W Unite 102,  Montréal, Que...   
3514         2151 Boul. Lapinière, Brossard, QC J4W 2T5   

                                              Name  
0          Métro Champ-de-Mars (Viger / Sanguinet)  
1                                  Clark / Ontario  
2                 de la Porte de Québec / St-André  
3                        St-Denis / de Maisonneuve  
4             Métro Sherbrooke (de Rigaud / Berri)  
...                                            ...  
3510                        Le Barrage - Brasseurs  
3511                    Firegrill Ville St-Laurent  
3512  Wild Wing Brasserie Sportive Greenfield Park  
3513                                    Entre-Deux  
3514                               Le White Rabbit  

[3515 rows x 5 columns]

In [48]:
geohash_6_df.to_csv('geohash_6_df.csv', index=False, encoding='utf-8-sig')

In [46]:
geohash_6_df = geohash_6_df[geohash_6_df['Geohash'].isin(geohash_list)]

In [ ]:
geohash_6_no_bikes_df = geohash_6_no_bikes_df[geohash_6_no_bikes_df['Geohash'].isin(geohash_list)]

In [51]:
geohash_6_no_bikes_df

Geohash Geohash_prec6            Type  \
0      f25dy        f25dyh        Citybike   
1      f25dv        f25dvu        Citybike   
2      f25dy        f25dym        Citybike   
3      f25dy        f25dyj        Citybike   
4      f25dv        f25dvy        Citybike   
...      ...           ...             ...   
3485   f25dv        f25dvp  Bar/Restaurant   
3486   f25ej        f25eju  Bar/Restaurant   
3487   f25ej        f25ejg  Bar/Restaurant   
3490   f25dv        f25dvx  Bar/Restaurant   
3491   f25dy        f25dy5  Bar/Restaurant   

                                          Address  \
0                                             NaN   
1                                             NaN   
2                                             NaN   
3                                             NaN   
4                                             NaN   
...                                           ...   
3485     5834 Park Ave,  Montréal, Quebec H2V 4H3   
3486        3560 Rue Masson, Montréal, QC H1X 1R9   
3487        2800 Rue Masson, Montréal, QC H1Y 1W7   
3490       500 Rue Rachel E, Montréal, QC H2J 2H2   
3491  390 Rue Saint-Jacques, Montréal, QC H2Y 1S1   

                                         Name  
0     Métro Champ-de-Mars (Viger / Sanguinet)  
1                             Clark / Ontario  
2            de la Porte de Québec / St-André  
3                   St-Denis / de Maisonneuve  
4        Métro Sherbrooke (de Rigaud / Berri)  
...                                       ...  
3485             Mezcaleros Tapas & Cocktails  
3486                          Café Bar U-Turn  
3487                                Quai No.4  
3490                        Name's on the way  
3491          Bord'Elle Boutique Bar & Eatery  

[1204 rows x 5 columns]

In [50]:
geohash_6_no_bikes_df.to_csv('geohash_6_no_bikes_df.csv', index=False, encoding='utf-8-sig')